ext/handlers_http.h

```diff
@@ -229,12 +301,27 @@ static inline void ddtrace_inject_distributed_headers_config(zend_array *array,
 
     zval headers;
     ZVAL_ARR(&headers, array);
+    smart_str stream_header = {0};
+    HashTable *header_keys = ddtrace_init_header_keys_set(array, format);
 
 #define ADD_HEADER(header, ...) \
-    if (key_value_pairs) { \
+    if (format == HEADER_MODE_KV_PAIRS) { \
         add_assoc_str_ex(&headers, ZEND_STRL(header), zend_strpprintf(0, __VA_ARGS__)); \
+    } else if (format == HEADER_MODE_ARRAY) { \
+        ddtrace_add_or_update_header_array(Z_ARRVAL(headers), header_keys, ZEND_STRL(header), zend_strpprintf(0, header ": " __VA_ARGS__)); \
```

Comment:
Can we provide the to-lower form directly in the macro to avoid lowercasing at runtime?
Then you also can use zend_hash_str_find() directly in ddtrace_add_or_update_header_array